pipelined/bevy_ui2/src/entity.rs

```diff
@@ -1,100 +1,41 @@
-use super::Node;
 use crate::{
-    render::UI_PIPELINE_HANDLE,
-    widget::{Button, Image},
-    CalculatedSize, FocusPolicy, Interaction, Style,
+    widget::{Button, ImageMode},
+    CalculatedSize, FocusPolicy, Interaction, Node, Style, UiColor, UiImage, CAMERA_UI,
 };
-use bevy_asset::Handle;
 use bevy_ecs::bundle::Bundle;
-use bevy_render::{
-    camera::{Camera, DepthCalculation, OrthographicProjection, VisibleEntities, WindowOrigin},
-    draw::Draw,
-    mesh::Mesh,
-    pipeline::{RenderPipeline, RenderPipelines},
-    prelude::Visible,
+use bevy_render2::{
+    camera::{Camera, DepthCalculation, OrthographicProjection, WindowOrigin},
+    view::VisibleEntities,
 };
-use bevy_sprite::{ColorMaterial, QUAD_HANDLE};
-use bevy_text::Text;
+use bevy_text2::Text;
 use bevy_transform::prelude::{GlobalTransform, Transform};
 
-#[derive(Bundle, Clone, Debug)]
+#[derive(Bundle, Clone, Debug, Default)]
 pub struct NodeBundle {
     pub node: Node,
     pub style: Style,
-    pub mesh: Handle<Mesh>, // TODO: maybe abstract this out
-    pub material: Handle<ColorMaterial>,
-    pub draw: Draw,
-    pub visible: Visible,
-    pub render_pipelines: RenderPipelines,
+    pub color: UiColor,
```

Comment:
Part of me wants to just merge UiColor and UiImage into `Node`. I really don't like wrapper structs, and we're adding these to all Nodes anyway. I think that would improve UX.

Reply:
Although I guess Node could be used in a lot of non-color + image contexts. I guess the `color.into()` pattern isn't terrible. We could also merge UiColor and UiImage into a single component and then used name fields, but that does make this less flexible (some widgets might want a color, but not an image). I guess lets leave this as is for now and we can revisit later if we can think of something better.